cdap-common/src/main/java/co/cask/cdap/common/service/RetryOnStartFailureService.java

```diff
@@ -58,7 +58,7 @@ public void run() {
         long delay = 0L;
         Service delegateService = service;
 
-        while (delay >= 0 && !isInterrupted()) {
+        while (delay >= 0 && !isInterrupted() && !stopped) {
```

Comment:
@gokulavasan brought up the point that using `Service#isRunning` can have issues, since in `RetryOnStartFailureService`'s `doStart` method starts the `startupThread`, and it might execute the `isRunning` method before service actually returns from the `doStart` method. Hence, `isRunning` would return false, leading to unexpected behavior.

Changed the code back to using a `stopped` flag.